### ISSUE TITLE
Write developer guidelines

### ISSUE THREAD

**Issue by FilippoOggionni** [tudat-team/tudat-developer-docs#13]
Write developer guidelines (as from this Slack [post](https://tudat.slack.com/archives/C02140PKDEJ/p1643931419264749)).

**Comment by FilippoOggionni** [comment]
Good point! I think including a page somewhere about the organization of the tudat ecosystem (all the repos, what they are meant for, etc...) is a great idea and very much needed. I already have some ideas about that and some material written in a README, so I will take care of that. This page should be the first things developers read / come across in the developer docs, in my opinion.

Question: do you think developers should read first the DevOps resources sections or first the checklist? In other words, do you think a developer needs to know the content from the resources _before_ being able to understand the checklist?

**Comment by gaffarelj** [comment]
> Question: do you think developers should read first the DevOps resources sections or first the checklist? In other words, do you think a developer needs to know the content from the resources before being able to understand the checklist?

I prefer having the checklist first, and resources second. But mentioning that the resources section exists at the beginning of the page could be useful.

**Comment by FilippoOggionni** [comment]
> But mentioning that the resources section exists at the beginning of the page could be useful.

Sounds good, I will do that!

Regarding the overview page, do you have already an idea of where it could be placed in the toctree?

**Comment by FilippoOggionni** [comment]
I added the following note, do you think it is enough?

![Screen Shot 2022-04-07 at 23 43 01](https://user-images.githubusercontent.com/57680167/162324857-55b44d1e-f5bc-419e-adde-4c0ae4eb2a5c.png)

**Comment by FilippoOggionni** [comment]
Changes have been implemented through 0ce93aa428365989f6b79ba440d52e55a10683cb (and published on the website at `latest`).

**Comment by gaffarelj** [comment]
> I added the following note, do you think it is enough?

Yes, I think nothing more would be needed, perfect for me :)

> Regarding the overview page, do you have already an idea of where it could be placed in the toctree?

I would expect to find it at the top of the [Software Development](https://tudat-developer.readthedocs.io/en/latest/primer/software.html) section (before the Build System and Developer Environment sections). Actually, I believe the [Developer Environment](https://tudat-developer.readthedocs.io/en/latest/primer/software/environment.html) section already covers most of what the overview page would describe. But I think the overview should go less into details, and give a more general... well... overview.

As a small note, I would also switch the sections in the following order: Software Development, Software Documentation, Dev Ops.

**Comment by DominicDirkx** [comment]
Hi all, sorry to be late to the party :) I have made a couple of very small modifications. Below a few suggestions/requests:

- It would be good to put contact info at the top of the page: if anyone is interested in contributing to Tudat, I think we should encourage them to contact us, if they have not yet.
- In the description here: https://tudat-developer.readthedocs.io/en/latest/primer/devops.html#checklist-for-developer
I think it's good to write that we assume that people are working from the tudat-bundle
- In step 2b-2e here: https://tudat-developer.readthedocs.io/en/latest/primer/devops.html#checklist-for-reviewer a (link to) a more elaborate desciption is needed
- This tip: https://github.com/tudat-team/tudat-developer-docs/blame/develop/source/primer/devops.rst#L83 is no clear to me, could you add how/why this is different from the other repos?
- These links: https://github.com/tudat-team/tudat-developer-docs/blame/develop/source/primer/devops.rst#L95 should be a bit higher, to make clear to a developer, who may not want to read everything ;), that we have support pages for this
- I see that the versioning guide: https://tudat-developer.readthedocs.io/en/latest/primer/devops/versioning.html#releasing-a-new-version-with-bumpversion only refers to the documentation. This should be extended to tudat and tudatpy, making clear the distinction (or lack thereof?) of version for the code (on Github) and the package (on Anaconda)
